wrapper/src/main/java/software/amazon/jdbc/plugin/failover2/FailoverConnectionPlugin.java

```diff
@@ -606,4 +621,69 @@ protected boolean canDirectExecute(final String methodName) {
         || methodName.equals(METHOD_IS_CLOSED)
         || methodName.equals(METHOD_ABORT));
   }
+
+  @Override
+  public Connection connect(
+      final String driverProtocol,
+      final HostSpec hostSpec,
+      final Properties props,
+      final boolean isInitialConnection,
+      final JdbcCallable<Connection, SQLException> connectFunc)
+      throws SQLException {
+
+    if (!ENABLE_CONNECT_FAILOVER.getBoolean(props)) {
+      return connectFunc.call();
```

Comment:
Shouldn't we be connecting using getVerifiedConnection in this case?
```suggestion
      return this.staleDnsHelper.getVerifiedConnection(isInitialConnection, this.hostListProviderService,
            driverProtocol, hostSpec, props, connectFunc);
```

Reply:
getVerifiedConnection() call already been made of a caller of connect() that is the same failover2 plugin. I don't think we need to execute it again.

Reply:
sorry are you saying that getVerifiedConnection has been previously called by another plugin or previously called by failover2?

Reply:
failover2 `connect()` is called, then `getVerifiedConnection()` is called.
https://github.com/aws/aws-advanced-jdbc-wrapper/blob/b84d66c8eb6361e7b036e085ff27f02c1e91025b/wrapper/src/main/java/software/amazon/jdbc/plugin/failover2/FailoverConnectionPlugin.java#L654

After that, if connection isn't established, a failover starts.

https://github.com/aws/aws-advanced-jdbc-wrapper/blob/b84d66c8eb6361e7b036e085ff27f02c1e91025b/wrapper/src/main/java/software/amazon/jdbc/plugin/failover2/FailoverConnectionPlugin.java#L664

And eventually it calls `pluginService.connect()`

https://github.com/aws/aws-advanced-jdbc-wrapper/blob/b84d66c8eb6361e7b036e085ff27f02c1e91025b/wrapper/src/main/java/software/amazon/jdbc/plugin/failover2/FailoverConnectionPlugin.java#L507

that will call failover2 `connect()` again. This call can be passed through to the next plugin since it's already been handled earlier.

Reply:
If I understand correctly this is explaining why we should use connectFunc.call instead of getVerifiedConnection when the props contain the INTERNAL_CONNECT_PROPERTY_NAME property, but this particular line is a different scenario: it is hit when connect failover is disabled. When connect failover is disabled we will never enter the failover logic below, so as the code currently is written getVerifiedConnection will never be called in this scenario